test/public-stats.test.js

```diff
@@ -627,6 +629,198 @@ describe('public-stats', () => {
+        const { rows: created } = await pgClient.query(
+          'SELECT * FROM daily_client_retrieval_stats'
+        )
+        assert.deepStrictEqual(created, [])
+        await updateDailyClientRetrievalStats(
+          pgClient,
+          committees,
+          findDealClients
+        )
+        const { rows: f0Stats } = await pgClient.query(
+          "SELECT day::TEXT,client_id,total,successful,successful_http FROM daily_client_retrieval_stats WHERE client_id = 'f0client'"
+        )
+        assert.strictEqual(f0Stats.length, 1)
+        assert.deepStrictEqual(f0Stats, [
+          { day: today, client_id: 'f0client', total: 2, successful: 2, successful_http: 2 }
+        ])
+        const { rows: f1Stats } = await pgClient.query(
+          "SELECT day::TEXT,client_id,total,successful,successful_http FROM daily_client_retrieval_stats WHERE client_id = 'f1client'"
+        )
+        // All measurments are successful and successful_http
+        assert.strictEqual(f1Stats.length, 1)
+        assert.deepStrictEqual(f1Stats, [
+          { day: today, client_id: 'f1client', total: 3, successful: 3, successful_http: 3 }
+        ])
+      })
+      it('aggregates overlapping per client stats', async () => {
+        /** @type {Measurement[]} */
+        const allMeasurements = [
+          // minerId f01test stores deals for both client f0 and f1
+          { ...VALID_MEASUREMENT, protocol: 'http', minerId: 'f01test' },
+          { ...VALID_MEASUREMENT, protocol: 'http', minerId: 'f01test' },
+
+          { ...VALID_MEASUREMENT, protocol: 'http', minerId: 'f1test' },
+          { ...VALID_MEASUREMENT, protocol: 'http', minerId: 'f1test' },
+          { ...VALID_MEASUREMENT, protocol: 'http', minerId: 'f1test' }
+        ]
+
+        // Separate the measurments into two groups, one for the client f0 and f1 and the other only for f1
+        const findDealClients = (minerId, _cid) => {
+          switch (minerId) {
+            case 'f01test':
+              return ['f0client', 'f1client']
+            case 'f1test':
+              return ['f1client']
+            default:
+              throw new Error('Unexpected minerId')
+          }
+        }
+        const committees = buildEvaluatedCommitteesFromMeasurements(allMeasurements)
+        await updateDailyClientRetrievalStats(
+          pgClient,
+          committees,
+          findDealClients
+        )
+        const { rows: f0Stats } = await pgClient.query(
+          "SELECT day::TEXT,client_id,total,successful,successful_http FROM daily_client_retrieval_stats WHERE client_id = 'f0client'"
+        )
+        assert.deepStrictEqual(f0Stats, [
+          { day: today, client_id: 'f0client', total: 2, successful: 2, successful_http: 2 }
+        ])
+        const { rows: f1Stats } = await pgClient.query(
+          "SELECT day::TEXT,client_id,total,successful,successful_http FROM daily_client_retrieval_stats WHERE client_id = 'f1client'"
```

Comment:
```suggestion
        const { rows } = await pgClient.query(
          'SELECT day::TEXT, client_id, total, successful, successful_http FROM daily_client_retrieval_stats ORDER BY client_id'
```

You could do one select query here where you do full table scan and just do one assert.

Reply:
Does it make much of a difference whether two queries are done or one query and then a filter operation to separate f1client from f0client?

Reply:
> Does it make much of a difference whether two queries are done or one query

I don't think it makes much difference in regards to performance or the end result (testing if logic was valid), but it does make code cleaner and simpler.

> ... and then a filter operation to separate f1client from f0client?

In the suggested change I've provided you would not filter the result, rather you would test the whole response, for example:

```javascript
const { rows } = await pgClient.query(
`SELECT 
     day::TEXT, 
     client_id, 
     total, 
     successful, 
     successful_http 
  FROM daily_client_retrieval_stats 
  ORDER BY client_id`)

assert.deepStrictEqual(f1Stats, [
  { day: today, client_id: 'f0client', total: 2, successful: 2, successful_http: 2 },
  { day: today, client_id: 'f1client', total: 5, successful: 5, successful_http: 5 }
])
```

Reply:
Thanks for your input. I addressed it in bc27e9a